packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -177,23 +216,26 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
       if (!_firstTap.isWithinTolerance(event, kDoubleTapSlop)) {
         // Ignore out-of-bounds second taps.
         return;
-      } else if (!_firstTap.hasElapsedMinTime() || !_firstTap.hasSameButton(event)) {
+      }
+      if (!_firstTap.hasElapsedMinTime() || !_firstTap.hasSameButton(event)) {
         // Restart when the second tap is too close to the first, or when buttons
         // mismatch.
         _reset();
-        return _trackFirstTap(event);
       }
     }
     _trackFirstTap(event);
   }
 
   void _trackFirstTap(PointerEvent event) {
     _stopDoubleTapTimer();
+
+    assert(event is PointerDownEvent);
```

Comment:
Not sure if this assertion is correct here but I think we're expecting a pointer down event?